### PR TITLE
[SYCLomatic][CMake] Allow definition options in cuda_compile,cuda_compile_fatbin

### PR DESCRIPTION
**Pull request by daiyaan-ahmed6**
Signed-off-by: Daiyaan Ahmed <daiyaan.ahmed@intel.com>

### CODE REVIEW

File: clang/lib/DPCT/MigrateScript/MigrateCmakeScript.cpp [hunk 1]
```diff
@@ -8,9 +8,9 @@
 #include "MigrateCmakeScript.h"
 #include "Diagnostics/Diagnostics.h"
 #include "ErrorHandle/Error.h"
-#include "UserDefinedRules/PatternRewriter.h"
 #include "FileGenerator/GenFiles.h"
 #include "MigrationReport/Statics.h"
+#include "UserDefinedRules/PatternRewriter.h"
```
Comment (tomflinda) [inline clang/lib/DPCT/MigrateScript/MigrateCmakeScript.cpp:13]: Can line13 be removed? it seems not be used in this file.

File: clang/lib/DPCT/MigrateScript/MigrateCmakeScript.cpp [hunk 2]
```diff
@@ -20,6 +20,7 @@
 #include <algorithm>
 #include <cstring>
 #include <fstream>
+#include <numeric>
```
Comment (tomflinda) [inline clang/lib/DPCT/MigrateScript/MigrateCmakeScript.cpp:23]: Line 23 seem  not be used in this file, can it be removed?

File: clang/lib/DPCT/MigrateScript/MigrateCmakeScript.cpp (hunk 1 above)
Comment (daiyaan-ahmed6) [reply clang/lib/DPCT/MigrateScript/MigrateCmakeScript.cpp:13]: Clang format automatically applies this. Line 13 is removed from line 10 and added to 13 to maintain alphabetical ordering of includes.

File: clang/lib/DPCT/MigrateScript/MigrateCmakeScript.cpp (hunk 2 above)
Comment (daiyaan-ahmed6) [reply clang/lib/DPCT/MigrateScript/MigrateCmakeScript.cpp:23]: std::accumulate required <numeric> inclusion

**Review by tomflinda — APPROVED** [review]
LGTM

**Review by ShengchenJ — APPROVED** [review]

**Pull request merged**